libs/utils/executor.py

```diff
@@ -252,6 +252,14 @@ def _setup_cpufreq(self, tc):
 
         self.target.cpufreq.set_all_governors(cpufreq['governor'])
 
+        if 'freqs' in cpufreq:
+            if cpufreq['governor'] != 'userspace':
+                raise ValueError('Must use userspace governor to set CPU freqs')
+            logging.info(r'%14s - CPU frequencies: %s',
+                    'CPUFreq', str(cpufreq['freqs']))
+            for cpu, freq in cpufreq['freqs'].iteritems():
```

Comment:
Ok, I did not noticed `cpufreq['freqs']` was a map... then that makes much more sense! ;-)